Fix bugs in the Sample Transmission Calculator

jclarkeSTFC opened the pull request:
**Description of work**
A few bug fixes and usability improvements for the Sample Transmission Calculator.

**Purpose of work**
<!-- Why has this work been done? If there is no linked issue please provide appropriate context for this work.
-->
Fixes #35011 and fixes #35586

<!-- If the original issue was raised by a user they should be named here. Do not leak email addresses
**Report to:** [user name]
-->

**Summary of work**
<!-- Please provide a short, high level description of the work that was done.
-->
- Added tool tip to edit box when using the Multiple option
- Added labels to the axes
- Add error if the bin width is larger than the max
- Highlighted error messages a bit more with a red box

**To test:**
See #35011 for how to reproduce.

<!-- Instructions for testing.
There should be sufficient instructions for someone unfamiliar with the application to test - unless a specific
reviewer is requested.
If instructions for replicating the fault are contained in the linked issue then it is OK to refer back to these.
-->

<!-- delete this if you added release notes
*This does not require release notes* because **fill in an explanation of why**
If you add release notes please save them as a separate file using the Issue or PR number as the file name. Check the file is located in the correct directory for your note(s).
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

jclarkeSTFC committed:
Add axes labels.

Axes labels on plot inside the Sample Transmission Calculator window.

jclarkeSTFC committed:
Error check on bin width.

Don't allow the bin width to be greater than the upper bin limit.

jclarkeSTFC committed:
Improve multiple bin info.

Add a tooltip to the edit box for entering the bins when using the Multiple option. When there aren't an odd number of values in the box then explain why this is a problem in the error message.

jclarkeSTFC committed:
Make errors more visible.

Now it draws a red box around the error message.

jclarkeSTFC committed:
Add release notes

jclarkeSTFC committed:
Update test cases.

The validation messages have changed, and the width is now checked relative to the max, so the tests needed updating. I also added a new test for checking the width > max validation message.

jhaigh0 commented on qt/python/mantidqtinterfaces/test/sample_transmission_calculator/stc_model_test.py:
```diff
@@ -98,7 +98,7 @@ def test_validate_histogram_single_max_lower_than_min(self):
             "binning_type": 0,
             "single_low": 1.0,
-            "single_width": 0.1,
+            "single_width": 0.0,
```
What was the reason for changing this value?

jclarkeSTFC replied:
The test name suggests that it's checking max < min, but it also had width > max, which would have generated two errors, so I added a separate test for checking the width > max case. The options for this case to make the test pass were to either change the width or change the expected error message, so I changed the width.

jhaigh0 commented on qt/python/mantidqtinterfaces/mantidqtinterfaces/SampleTransmissionCalculator/stc_view.py:
```diff
@@ -19,13 +19,21 @@ def __init__(self, parent=None):
         fig = Figure()
         self.axes = fig.add_subplot(111)
         self.plot_frame = FigureCanvas(fig)
```
When resizing to a small size, the axis labels can run off the edge of the plot area and become hidden. Giving the figure a tight layout is the simplest way to fix this.

```suggestion
        fig = Figure()
        fig.set_layout_engine(layout="tight")
        self.axes = fig.add_subplot(111)
        self.plot_frame = FigureCanvas(fig)
```

jhaigh0 reviewed (changes requested):
Nice improvements to the feedback of the interface :)

One further thing I wondered about was changing where the 'expanding areas' of the gui are for single input.
![image](https://github.com/mantidproject/mantid/assets/35813666/9de3824a-ecab-4195-98af-702779439416)

At the moment when you expand the window it adds space inbetween say `Width:` and the width box. I think these expansions should happen inbetween the previous box and the next lable. So labels stay attached to there respective boxes.

jclarkeSTFC committed:
Update release notes

Co-authored-by: Jonathan Haigh <35813666+jhaigh0@users.noreply.github.com>

jclarkeSTFC committed:
Give figure a tight layout

Prevents axes running off the edge of the plot area.

Co-authored-by: Jonathan Haigh <35813666+jhaigh0@users.noreply.github.com>

jclarkeSTFC committed:
Add info from tooltip to docs.

jclarkeSTFC committed:
Improve ui resize behaviour.

Previously if you changed the window width then the new space would appear between the box labels and the input boxes. I've added spacers so the labels stay next to the input boxes.

jhaigh0 reviewed (approved):
Thanks for making the changes 👍
